### PR TITLE
feat(core): Adds opencv add operation

### PR DESCRIPTION
**Pull request by JLLeitschuh**

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
Don't approve this PR yet. I'm going to clean up the commit history.

**Comment by JLLeitschuh** [comment]
This is ready to go. Finally. Only took ~9 hours in total to get the build system to be happy.

File: build.gradle [hunk 1]
```diff
@@ -25,8 +25,10 @@ jacocoTestReport {
         html.enabled = true
     }
 }
+
```
Comment (ThomasJClark) [inline build.gradle]: Get rid of the commit noise.

File: src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    public Socket[] createInputSockets(EventBus eventBus) {
+        return new Socket[]{new Socket<>(eventBus, aHint), new Socket<>(eventBus, bHint)};
+    }
+
+    @Override
+    public Socket[] createOutputSockets(EventBus eventBus) {
+        System.out.println("entering createOutputSockets");
+        Socket<Mat> output = new Socket<>(eventBus, sumHint);
+        System.out.println("middle createOutputSockets");
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java]: Get rid of these printlns

File: src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        output.setValue(new Mat());
```
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java]: There's no need to call `setValue()`.  `SocketHint` has a constructor that takes an initial value.

File: src/test/java/edu/wpi/grip/core/operations/opencv/AddOperationTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        int sz[] = {100, 100, 100};
+
+        a.setValue(new Mat (3, sz, opencv_core.CV_8U, Scalar.all(1)));
+        b.setValue(new Mat (3, sz, opencv_core.CV_8U, Scalar.all(2)));
+
+        //When
+        addition.perform(inputs, outputs);
+
+        //Then
+        Mat expectedResult = new Mat (3, sz, opencv_core.CV_8U, Scalar.all(3));
```
Comment (ThomasJClark) [inline src/test/java/edu/wpi/grip/core/operations/opencv/AddOperationTest.java]: Fix the formatting issues (Ctrl+Alt+L on most platforms in IntelliJ)

File: src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java (hunk 3 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java]: You explicitly told me not to use that.... Gah....

File: src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java (hunk 2 above)
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java]: Didn't see those. Thanks for noticing.

File: src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java (hunk 3 above)
Comment (ThomasJClark) [inline src/main/java/edu/wpi/grip/core/operations/opencv/AddOperation.java]: My bad, it's `Socket`, not `SocketHint` that has the constructor you should use.

File: src/test/java/edu/wpi/grip/core/operations/opencv/AddOperationTest.java (hunk 4 above)
Comment (JLLeitschuh) [inline src/test/java/edu/wpi/grip/core/operations/opencv/AddOperationTest.java]: Oooh.. Very nice!

**Comment by JLLeitschuh** [comment]
Fixed it up from your suggestions.

**Pull request merged**